compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/calls/ResolutionStages.kt

```diff
@@ -153,8 +155,28 @@ internal object MapArguments : ResolutionStage() {
     override suspend fun check(candidate: Candidate, callInfo: CallInfo, sink: CheckerSink, context: ResolutionContext) {
         val symbol = candidate.symbol as? FirFunctionSymbol<*> ?: return sink.reportDiagnostic(HiddenCandidate)
         val function = symbol.fir
+        val simpleFunction = function as? FirSimpleFunction
+
+        var arguments = callInfo.arguments
+        // If this is an overloading indexed access operator, it could have default values in the middle.
+        // For proper argument mapping, wrap the last one, which is supposed to be the updated value, as a named argument.
+        if (simpleFunction?.isOperator == true &&
```

Comment:
Good shot! Please, consider moving this logic into `mapArguments`. Not quite sure but think it could be more logical...

Reply:
No problem! Moved to `mapArguments`.